rllib/evaluation/tests/test_agent_collector.py

```diff
@@ -332,6 +332,19 @@ def test_view_requirement_with_shfit_step(self):
         # check if the padding in the beginning is correct
         check(sample_batch["prev_obses"][0], np.ones((3, 1)) * expected_obses[0])
 
+    def test_begin_with_inference(self):
+        # Test if AC is able to build for inference immediately
+        obs_space = gym.spaces.Box(-np.ones(4), np.ones(4))
+        view_reqs = {
+            SampleBatch.T: ViewRequirement(SampleBatch.T),
+            SampleBatch.REWARDS: ViewRequirement("reward", space=obs_space),
+            "prev_reward": ViewRequirement("reward", shift=-1),
+        }
+
+        ac = AgentCollector(view_reqs=view_reqs, is_policy_recurrent=True)
```

Comment:
wait this test not using AgentCollector properly. If there is no data added to the buffer (neither with `ac.add_init_obs` nor `ac.add_action_reward_next_obs`) how is it suppose to build a batch for inference. I agree that the error is not informative right now (IndexError) but I don't think you should be allowed to call `ac.build_for_inference()` without having anything in the AgentCollector buffers.

Reply:
for example look at `_simulate_env_steps` on how timestep infos are added to the collector.

Reply:
Makes sense. The problem surfaces when running PPO with LSTM. I just checked and add_init_obs is called before the call that errors out.
I'll come up with a unittest that is essentially a repro of what happens there and we can see again if it's intended usage of the agent collector.